controllers/placementapi_controller.go

```diff
@@ -1181,7 +1183,9 @@ func (r *PlacementAPIReconciler) ensureDeployment(
 		return ctrl.Result{}, err
 	}
 
-	if instance.Status.ReadyCount > 0 || *instance.Spec.Replicas == 0 {
+	// We need to check generation with a similar pattern to k8s,
+	// just in case when the status is recreated or the etcd database needs to be recreated
+	if (instance.Status.ReadyCount == *instance.Spec.Replicas && deployment.Generation <= deployment.Status.ObservedGeneration) || *instance.Spec.Replicas == 0 {
```

Comment:
I see the comment on the etcd database recreation that justify having `deployment.Generation <= deployment.Status.ObservedGeneration`: do you have more insights about this?

Reply:
I see for instance [1][2] but I'm not sure there's some reading I can do that describes the edge case we were discussing.

[1] https://github.com/kubernetes/kubectl/blob/master/pkg/polymorphichelpers/rollout_status.go#L75
[2] https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/util/deployment_util.go#L712

Reply:
Unfortunatly IMO there is not any good reading about this. My comment is based on some comments in pr's and stack overflow which I saw and discussions. 
Only docs that mention this pattern are without any additional info except that Generation <= ObservedGeneration is used there. eg https://github.com/kubernetes/design-proposals-archive/blob/acc25e14ca83dfda4f66d8cb1f1b491f26e78ffe/api-machinery/garbage-collection.md?plain=1#L207 or https://github.com/kubernetes/design-proposals-archive/blob/acc25e14ca83dfda4f66d8cb1f1b491f26e78ffe/api-machinery/garbage-collection.md?plain=1#L168 . Also there is the same question asked on k8s forum https://discuss.kubernetes.io/t/in-what-case-of-deployment-deployment-generation-deployment-status-observedgeneration-is-true/24655 without answers

Reply:
the only logic behind this check that I am able to see is covering a cached read